src/libstd/sys/unix/stack_overflow.rs

```diff
@@ -83,16 +83,19 @@ mod imp {
     // were originally supposed to do.
     //
     // This handler currently exists purely to print an informative message
-    // whenever a thread overflows its stack. When run the handler always
-    // un-registers itself after running and then returns (to allow the original
-    // signal to be delivered again). By returning we're ensuring that segfaults
-    // do indeed look like segfaults.
+    // whenever a thread overflows its stack. We then abort to exit with an
+    // error (and dump core if necessary), but to avoid a misleading SIGSEGV
```

Comment:
> though in that case it didn't have the message because of lack of stack probes #16012 not because of output redirection

The lack of stack probes would result in no SIGABRT too, so my point still stands.

> for instance if something captures the exit code but not stderr

That’s not something we should be worried about in technical documentation, right?

> and dump core if necessary

NIT: its not us dumping the core, nor do we do anything beyond raising a SIGABRT to make the coredump happen. IMO this is confusing and should be removed too.